pkg/config/legacy/converter.go

```diff
@@ -283,6 +338,30 @@ func extractURLAPIKeys(agentConfig Config, configConverter *config.LegacyConfigC
 		}
 	}
 	configConverter.Set("additional_endpoints", additionalEndpoints)
+
```

Comment:
I might be missing something but why isn't this piece of code part of `extractProcesAgentConfig` ?

Reply:
The way we handle `additional_endpoints` on the process agent is similar to how it's done for the main agent. We need to have access to the api keys, which is parsed in `extractURLAPIKeys`. 

Our logic is quite similar to the above core-agent `additional_endpoints` handling so this seems like the most logical place to have the logic :)

Reply:
I feel it'll be confusing in the future when someone will expect all the process agent stuff to be coupled together. If you just need the api keys you can always `.Get("api_key")` and it will be fine. If you need API keys from the `additional_endpoints`, my opinion is that it's better to either duplicate that logic or factor it out into a helper which gets called from both places.

If you feel strongly about your current model and nobody else comments on it, leave it.

Reply:
I agree with @gbbr, makes more sense to me to have this with the rest of the process agent code

Reply:
Fixed.